exercise/views.py

```diff
@@ -144,9 +146,23 @@ def post(self, request, *args, **kwargs):
             self.submission_check(True, request)
         )
         if submission_allowed:
-            new_submission = Submission.objects.create_from_post(
-                self.exercise, students, request)
-            if new_submission:
+            try:
+                new_submission = Submission.objects.create_from_post(
+                    self.exercise, students, request)
+            except ValueError as error:
+                messages.error(request,
+                    format_lazy(
+                    _("The submission could not be saved due to malformed post data. "
```

Comment:
You have modified or added some translated strings, so don't forget to add Finnish translations. `manage.py makemessages`, update the file `locale/fi/LC_MESSAGES/django.po` and `manage.py compilemessages` etc.